### PR TITLE
fix(ext/node): rewrite Pipe.open(fd) to use raw fd ops

### PR DESCRIPTION
**Pull request by bartlomieju**
## Summary

Rewrites `Pipe.open(fd)` to use the `NodeFsState` fd-based I/O infrastructure (from #33039) instead of creating Deno resources.

- **New op: `op_node_register_fd`** -- registers a raw OS fd in `NodeFsState` so existing fd-based read/write/close ops work on it. Supports both Unix (`from_raw_fd`) and Windows (`DuplicateHandle` + `from_raw_handle`).
- **Replace `FileStreamConn`** (resource-based, with EAGAIN polling) with `FdStreamBase` that uses `op_node_fs_read_deferred`/`op_node_fs_write_deferred` directly on the raw fd.
- **`Pipe.open(fd)` now works on Windows** (was `notImplemented`).
- **Remove `op_net_unix_stream_from_fd`** and `is_socket_fd` from `ext/net` -- no longer needed.

This is a step toward implementing `net.Socket({ fd })` and fixing the Git Bash TTY panic (#33131).

## Test plan

- [x] `./x test-spec stream_wrap` -- all 4 tests pass
- [x] `./x test-unit http` -- passes
- [x] `./x test-node raw_fd` -- passes
- [x] Lint and format clean
- [ ] Full CI

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ext/node/polyfills/internal_binding/pipe_wrap.ts [hunk 1]
```diff
@@ -74,52 +74,59 @@ export enum socketType {
...
+      this.#closed = true;
+      op_node_fs_close(this.#fd);
+    }
+  }
+
+  ref(): void {
+    if (this.#pendingRead) {
+      core.refOpPromise(this.#pendingRead);
+    }
+  }
```
Comment (bartlomieju) [inline ext/node/polyfills/internal_binding/pipe_wrap.ts:130]: This seems wrong - if there's no pending read it should be probably marked that whole `FdStreamBase` is "refed"

File: ext/node/polyfills/internal_binding/pipe_wrap.ts [hunk 2]
```diff
@@ -74,52 +74,59 @@ export enum socketType {
...
+
+  unref(): void {
+    if (this.#pendingRead) {
+      core.unrefOpPromise(this.#pendingRead);
+    }
```
Comment (bartlomieju) [inline ext/node/polyfills/internal_binding/pipe_wrap.ts:136]: Similarly here, so when a read is started it's immediately unrefed

File: tests/specs/node/pipe_open_fd/__test__.jsonc
Comment (bartlomieju) [inline tests/specs/node/pipe_open_fd/__test__.jsonc:1]: These tests should be brought back

File: tests/unit_node/net_pipe_fd_test.ts
Comment (bartlomieju) [inline tests/unit_node/net_pipe_fd_test.ts:1]: These test are completely useless and should be removed

File: tests/specs/node/pipe_open_fd/__test__.jsonc
Comment (bartlomieju) [reply tests/specs/node/pipe_open_fd/__test__.jsonc:1]: Also Windows version of these tests should be added

**Comment by bartlomieju** [comment]
## Follow-up: child_process stdio should use real fds

`ext/node/polyfills/internal/child_process.ts` currently uses a `StreamResource` class that wraps Deno resource IDs for child stdio pipes:

```js
this.stdout = new Socket({
  handle: new Pipe(socketType.SOCKET, new StreamResource(stdioRids.stdoutRid)),
  ...
});
```

Once `net.Socket({ fd })` lands (next PR in this stack), this should be refactored to:
1. Have `internals.getStdioRids` return real OS fds instead of rids
2. Replace with `new Socket({ fd })` which goes through `_createHandle(fd)` -> `Pipe.open(fd)` -> `FdStreamBase`

This would eliminate the `StreamResource` class and align with how Node.js handles child process stdio. Also unblocks re-landing #32959 (numeric fds in child_process stdio array).

File: ext/node/ops/fs.rs [hunk 3]
```diff
@@ -852,6 +852,86 @@ pub async fn op_node_rmdir(
...
+  if fd < 0 {
+    return Err(ebadf());
+  }
+
+  // SAFETY: The caller is responsible for passing a valid fd that they own.
+  // The fd will be owned by the File from this point on.
+  let std_file = unsafe { StdFile::from_raw_fd(fd) };
+  let file: Rc<dyn deno_io::fs::File> =
+    Rc::new(deno_io::StdFileResourceInner::file(std_file, None));
+  state.borrow_mut::<NodeFsState>().open_fds.insert(fd, file);
```
Comment (miracatbot) [inline ext/node/ops/fs.rs:899]: `op_node_register_fd()` takes ownership of the passed fd/handle, but this inserts into `NodeFsState.open_fds` unconditionally. If the same numeric fd is already tracked, `HashMap::insert()` will replace the previous owned `File` object without going through the normal close path for that registration. That makes the ownership semantics fragile for repeated `Pipe.open(fd)` / future reuse of this op, and on Windows it can also leak the previous CRT-fd bookkeeping because `op_node_fs_close()` is the place that calls `libc::close(fd)` for tracked entries.

Could we either reject already-registered fds here, or explicitly remove+close the old entry before replacing it? I’d also add a regression test for duplicate registration / reopening the same fd number.
